pygaggle/run/evaluate_document_ranker.py

```diff
@@ -0,0 +1,187 @@
+from pygaggle.model import (SimpleBatchTokenizer,
+                            T5BatchTokenizer,
+                            RerankerEvaluator,
+                            metric_names,
+                            MsMarcoWriter)
+from pygaggle.data import MsMarcoDataset
+from pygaggle.settings import MsMarcoSettings
+
+
+SETTINGS = MsMarcoSettings()
+METHOD_CHOICES = ('transformer', 'bm25', 't5', 'seq_class_transformer',
+                  'random')
+
+
+class DocumentRankingEvaluationOptions(BaseModel):
+    task: str
+    dataset: Path
+    index_dir: Path
+    method: str
+    model: str
+    split: str
+    batch_size: int
+    seg_size: int
+    seg_stride: int
+    aggregate_method: str
+    device: str
+    is_duo: bool
+    from_tf: bool
+    metrics: List[str]
+    model_type: Optional[str]
+    tokenizer_name: Optional[str]
+
+    @validator('task')
+    def task_exists(cls, v: str):
+        assert v in ['msmarco']
+
+    @validator('dataset')
+    def dataset_exists(cls, v: Path):
+        assert v.exists(), 'data directory must exist'
+        return v
+
+    @validator('index_dir')
+    def index_dir_exists(cls, v: Path):
+        assert v.exists(), 'index directory must exist'
+        return v
+
+    @validator('model')
+    def model_sane(cls, v: str, values, **kwargs):
+        method = values['method']
+        if method == 'transformer' and v is None:
+            raise ValueError('transformer name or path must be specified')
+        return v
+
+    @validator('tokenizer_name')
+    def tokenizer_sane(cls, v: str, values, **kwargs):
+        if v is None:
+            return values['model']
+        return v
+
+
+def construct_t5(options: DocumentRankingEvaluationOptions) -> Reranker:
+    device = torch.device(options.device)
+    model = T5ForConditionalGeneration.from_pretrained(options.model,
+                                                       from_tf=options.from_tf).to(device).eval()
+    tokenizer = AutoTokenizer.from_pretrained(options.model_type)
+    tokenizer = T5BatchTokenizer(tokenizer, options.batch_size)
+    return T5Reranker(model, tokenizer)
+
+
+def construct_transformer(options:
+                          DocumentRankingEvaluationOptions) -> Reranker:
+    device = torch.device(options.device)
+    model = AutoModel.from_pretrained(options.model,
+                                      from_tf=options.from_tf).to(device).eval()
+    tokenizer = SimpleBatchTokenizer(AutoTokenizer.from_pretrained(
+        options.tokenizer_name),
+        options.batch_size)
+    provider = CosineSimilarityMatrixProvider()
+    return UnsupervisedTransformerReranker(model, tokenizer, provider)
+
+
+def construct_seq_class_transformer(options: DocumentRankingEvaluationOptions
+                                    ) -> Reranker:
+    try:
+        model = AutoModelForSequenceClassification.from_pretrained(
+            options.model, from_tf=options.from_tf)
+    except AttributeError:
```

Comment:
do we still need this?

Reply:
I am not sure about this model. We copied these part from msmarco-passage-experiment. Remove the `except` part?

Reply:
Yes, except can be removed.